cli/args.go

```diff
@@ -290,11 +299,24 @@ func filterTerragruntArgs(args []string) []string {
 	return out
 }
 
+// isDeprecatedOption checks if provided option is deprecated, and returns its substitution
+// TODO: ideally, it would be better to make this return (string, err)
+func isDeprecatedOption(optionName string) string {
+	newOption, deprecated := DEPRECATED_ARGUMENTS[optionName]
+	if deprecated {
+		logger := util.CreateLogEntry("", "debug")
+		logger.Warnf("Command line option %s is deprecated, please consider using %s", optionName, newOption)
+		return newOption
+	}
+	return optionName
+}
+
 // Find a boolean argument (e.g. --foo) of the given name in the given list of arguments. If it's present, return true.
 // If it isn't, return defaultValue.
 func parseBooleanArg(args []string, argName string, defaultValue bool) bool {
 	for _, arg := range args {
 		if arg == fmt.Sprintf("--%s", argName) {
+			argName = isDeprecatedOption(argName)
```

Comment:
here we need to run `isDeprecatedOption()` to give us deprecation warning. Variable itself is not used.

Reply:
If the returned value isn't used, we shouldn't set it on the calling side. Otherwise, it makes it _seem_ like it's used, and may confuse maintainers (like me!) if that's intentional or not.

Reply:
I reworked the `isDeprecatedOption()`, so it would be more obvious.